server/android/src/android.rs

```diff
@@ -2,167 +2,89 @@
+                init_sql: Some(format!("PRAGMA temp_store_directory = '{}';", cache_dir)),
+            },
+            // sync settings need to be configured at runtime
+            sync: None,
+            logging: None,
+        };
 
         // run server in background thread
         let thread = thread::spawn(move || {
-            actix_web::rt::System::new().block_on(async move {
-                let settings = Settings {
-                    server: ServerSettings {
-                        port,
-                        danger_allow_http: true,
-                        debug_no_access_control: false,
-                        cors_origins: vec!["http://localhost".to_string()],
-                        base_dir: Some(files_dir.to_str().unwrap().to_string()),
-                        machine_uid: Some(android_id),
-                    },
-                    database: DatabaseSettings {
-                        username: "n/a".to_string(),
-                        password: "n/a".to_string(),
-                        port: 0,
-                        host: "n/a".to_string(),
-                        database_name: db_path.to_string_lossy().to_string(),
-                        // See https://github.com/openmsupply/remote-server/issues/1076
-                        init_sql: Some(format!("PRAGMA temp_store_directory = '{}';", cache_dir)),
-                    },
-                    // sync settings need to be configured at runtime
-                    sync: None,
-                    logging: None,
-                };
-                let _ = start_server(settings, off_switch_receiver).await;
-            });
+            actix_web::rt::System::new()
+                .block_on(start_server(settings, off_switch_receiver))
+                .unwrap();
         });
 
         let mut bucket = SERVER_BUCKET.lock().unwrap();
-        bucket.insert(RunningServerHandle { thread, off_switch })
+        *bucket = Some(ServerBucket { off_switch, thread });
     }
 
     #[no_mangle]
-    pub unsafe extern "C" fn Java_org_openmsupply_client_RemoteServer_stopServer(
+    pub unsafe extern "C" fn Java_org_openmsupply_mobile_RemoteServer_stopServer(
         _: JNIEnv,
         _: JClass,
-        handle: jlong,
-    ) -> jshort {
-        let mut bucket = SERVER_BUCKET.lock().unwrap();
-        let handle = match bucket.map.remove(&handle) {
-            Some(handle) => handle,
-            None => return -1,
-        };
-        match futures::executor::block_on(handle.off_switch.send(())) {
-            Ok(_) => {}
-            Err(_) => return -1,
-        }
-        match handle.thread.join() {
-            Ok(_) => {}
-            Err(_) => return -1,
-        }
-
-        0
+    ) {
+        let ServerBucket { off_switch, thread } = SERVER_BUCKET.lock().unwrap().take().unwrap();
+        futures::executor::block_on(off_switch.send(())).unwrap();
+        thread.join().unwrap();
```

Comment:
Maybe handle some of the unwraps (Mutex unwrap is fine). Probably not needed but seems to better to do anyway in case we get into a weird state...